Config for Carbine conventional

KnowingNova opened the pull request.

KnowingNova committed:
Config for Carbine conventional

KnowingNova committed:
Convential Carbine

I hope it works for others

Danice123 commented:
I sure this is not a big issue, but you should probably make the pull request after you have finished totally on the project, just in case one of the reviewers looks at it prematurely and hits the merge button or something. :)

Danice123 commented on LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_LWConventionalWeapons.uc:
```diff
@@ -105,6 +117,73 @@ static function X2DataTemplate CreateTemplate_SMG_Conventional()
+	Template.ItemCat = 'weapon';
+	Template.WeaponCat = 'rifle';
+	Template.WeaponTech = 'conventional';
+	Template.strImage = "img:///UILibrary_SMG.conventional.LWConvSMG_Base";
```
Use
`Template.strImage = "img:///" $ default.Carbine_Conventional_ImagePath;`
here

Danice123 commented on LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_LWConventionalWeapons.uc:
```diff
@@ -9,6 +9,7 @@ class X2Item_LWConventionalWeapons extends X2Item config(LW_WeaponPack);
 // Variables from config - GameData_WeaponData.ini
 // ***** Damage arrays for attack actions  *****
 var config WeaponDamageValue SMG_CONVENTIONAL_BASEDAMAGE;
+var config WeaponDamageValue Carbine_CONVENTIONAL_BASEDAMAGE;
```
It would be nice if you followed the conventions of using all caps for these variables

Danice123 commented on LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_LWConventionalWeapons.uc:
```diff
@@ -21,6 +22,16 @@ var config int SMG_CONVENTIONAL_TRADINGPOSTVALUE;
+var config int Carbine_CONVENTIONAL_ICLIPSIZE;
+var config int Carbine_CONVENTIONAL_ISOUNDRANGE;
+var config int Carbine_CONVENTIONAL_IENVIRONMENTDAMAGE;
+var config int Carbine_CONVENTIONAL_ISUPPLIES;
```
Remove this line, its unnessesary

Danice123 commented on LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Ability_CarbineAbilities.uc:
```diff
@@ -0,0 +1,70 @@
+	dependson (XComGameStateContext_Ability) config(LW_WeaponPack);
+	
+// ***** Mobility bonuses for Carbines
+var config int Carbine_CONVENTIONAL_MOBILITY_BONUS;
```
It would be nice if you followed the conventions of using all caps for these variables

Danice123 commented on LWOTC_WeaponPack/Src/LWOTC_WeaponPack/Classes/X2Item_LWConventionalWeapons.uc:
```diff
@@ -21,6 +22,16 @@ var config int SMG_CONVENTIONAL_TRADINGPOSTVALUE;
+var config int Carbine_CONVENTIONAL_ISOUNDRANGE;
+var config int Carbine_CONVENTIONAL_IENVIRONMENTDAMAGE;
+var config int Carbine_CONVENTIONAL_ISUPPLIES;
+var config int Carbine_CONVENTIONAL_TRADINGPOSTVALUE;
```
And this one

TNychka reviewed (changes requested):
Please add Laser, Mag, Coil and beam weapons to this PR as well. It should mostly be a copy paste of what you have.

TNychka commented on LWOTC_WeaponPack/Localization/SMGs/XComGame.int:
```diff
@@ -21,4 +21,11 @@ FriendlyName="Scramgun"
+FriendlyName="Carbine"
+FriendlyNamePlural="Carbines"
+BriefSummary="Carbine!!! Submachine guns offer a lighter, more concealable alternative to the standard assault rifle, although they lack the same punch and are less accurate at long range."
+TacticalText="<Bullet/>Carbine!!! Firing the primary weapon ends the soldier's turn, in the absence of special training.\n<Bullet/> Use upgrades recovered from ADVENT enemies to improve the SMG's tactical effectiveness. \n<Bullet/> SMG-style weapons offer bonus mobility, close-quarters accuracy and concealment relative to assault rifles, but they do less damage, have less ammo, and suffer accurac ...
```
Can you clean this up? You get full creative freedom to write what you want here.

TNychka commented on LWOTC_WeaponPack/Localization/SMGs/XComGame.int:
```diff
@@ -21,4 +21,11 @@ FriendlyName="Scramgun"
+FriendlyNamePlural="Carbines"
+BriefSummary="Carbine!!! Submachine guns offer a lighter, more concealable alternative to the standard assault rifle, although they lack the same punch and are less accurate at long range."
+TacticalText="<Bullet/>Carbine!!! Firing the primary weapon ends the soldier's turn, in the absence of special training.\n<Bullet/> Use upgrades recovered from ADVENT enemies to improve the SMG's tactical effectiveness. \n<Bullet/> SMG-style weapons offer bonus mobility, close-quarters accuracy and concealment relative to assault rifles, but they do less damage, have less ammo, and suffer accurac ...
+AbilityDescName="SMG"
```
Also this shouldn't be SMG

TNychka reviewed (dismissed):
Good start, but it would be great if you could do all of the carbines at once. Don't worry about how long it takes! :)

KnowingNova committed:
Carbine Capital

Carbine values made capital

KnowingNova committed:
Working aim bonus + spawn in old save

Almost all descriptions are working
The aim and mobilety bonus is working
the armory show the rigth stads
and in a clean save the weapon wil know spawn

TNychka commented on LWOTC_WeaponPack/Localization/SMGs/XComGame.int:
```diff
@@ -26,6 +26,6 @@ AbilityDescName="SMG"
-BriefSummary="Carbine!!! Submachine guns offer a lighter, more concealable alternative to the standard assault rifle, although they lack the same punch and are less accurate at long range."
-TacticalText="<Bullet/>Carbine!!! Firing the primary weapon ends the soldier's turn, in the absence of special training.\n<Bullet/> Use upgrades recovered from ADVENT enemies to improve the SMG's tactical effectiveness. \n<Bullet/> SMG-style weapons offer bonus mobility, close-quarters accuracy and concealment relative to assault rifles, but they do less damage, have less ammo, and suffer accurac ...
-AbilityDescName="SMG"
+BriefSummary="Carbine offer a lighter and more accurat alternative to the standard smg, although they lack the same concealabilety."
```
Carbines offer a heavier and more accurate alternative to the standard SMG, although they lack the same concealability.
